ci/infra/pn-monitoring-cicd.yaml

```diff
@@ -0,0 +1,482 @@
+      Runtime: nodejs22.x
+      Handler: index.lambda_handler
+      Code: 
+        ZipFile: |
+          const { ECRClient, DescribeRepositoriesCommand, DescribeRegistryCommand, DescribeImagesCommand, StartImageScanCommand } = require("@aws-sdk/client-ecr");
+
+          const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+          async function getAllElements(func, param, ...args) {
+            let nextToken = undefined;
+            let first = true
+            let elements = []
+            while(first || nextToken) {
+              first = false;
+              let result = await func(...args, nextToken)
+              nextToken = result.nextToken
+              elements = elements.concat(result[param])
+            }
+            return elements
+          }
+
+          async function getRegistryId(awsClient) {
+            const input = {};
+            const command = new DescribeRegistryCommand(input);
+            const response = await awsClient.send(command);
+            return response.registryId;
+          }
+
+          async function getRepositoriesList(awsClient, registryId, nextToken){
+            const input = { // DescribeRepositoriesRequest
+              registryId: registryId,
+              maxResults:  100,
+            };
+            nextToken ? input.nextToken = nextToken : null
+            const command = new DescribeRepositoriesCommand(input);
+            const response = await awsClient.send(command);
+            return response
+          }
+
+
+          async function getRepositoryImagesDescription(awsClient, registryId, repositoryName, nextToken){
+            const input = { // ListImagesRequest
+              registryId: registryId,
+              repositoryName: repositoryName, // required
+              maxResults: 100,
+              filter: { // ListImagesFilter
+                tagStatus: "TAGGED"
+              },
+            };
+            nextToken ? input.nextToken = nextToken : null
+            const command = new DescribeImagesCommand(input);
+            const response = await awsClient.send(command);
+            return response
+          }
+
+          async function startImageScan(awsClient, registryId, repositoryName, imageId){
+            const input = { // StartImageScanRequest
+              registryId: registryId,
+              repositoryName: repositoryName, // required
+              imageId: imageId // ImageIdentifier
+            };
+            const command = new StartImageScanCommand(input);
+            const response = await awsClient.send(command);
+            return response
+          }
+
+          function extractLatestImages(images) {
+            const stable = [];
+            const rc = [];
+            const snapshot = [];
+
+            for (const img of images) {
+              const tags = img.imageTags || [];
+              for (const tag of tags) {
+                if (/(?:.*-)?\d+\.\d+\.\d+$/.test(tag)) {
```

Comment:
@ViVeniero nice to have: le regexp definite all'inizio come costanti sarebbero più leggibili